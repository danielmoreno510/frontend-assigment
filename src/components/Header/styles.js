const styles = {
  container: {
    background: "#1b80c6",
    display: "flex",
    color: "white",
    justifyContent: "space-between",
    fontSize: "14px"
  },
  sectionsFlex: {
    display: "flex",
    alignItems: "center"
  },
  containerLogo: {
    padding: "12px 48px 12px 32px"
  },
  logo: {
    height: "44px"
  },
  searchContainer: {
    display: "flex",
    marginRight: "48px"
  },
  search: {
    height: "24px",
    padding: "3px",
    background: "white",
    borderTopLeftRadius: "10px",
    marginRight: "2px"
  },
  input: {
    width: "260px",
    height: "28px",
    border: "0px",
    borderBottomRightRadius: "10px"
  },
  home: {
    height: "22px",
    marginRight: "8px"
  },
  settings: {
    alignItems: "center",
    display: "flex",
    padding: "0px 26px 0px 26px",
    borderLeft: "1px solid",
    margin: "10px 0px 10px 0px"
  },
  imageSettings: {
    width: "166px"
  }
};

export default styles;
