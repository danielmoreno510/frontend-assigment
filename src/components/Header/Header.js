import React from "react";

import styles from './styles'
import logo from '../../assets/images/logo.png'
import search from '../../assets/images/search.png'
import home from '../../assets/images/home.png'
import settings from '../../assets/images/settings.png'

const Header = () => {
  return (
    <div style={styles.container}>
      <div style={styles.sectionsFlex}>
        <div style={styles.containerLogo}>
          <img style={styles.logo} src={logo}/>
        </div>
        <div style={styles.searchContainer}>
          <img style={styles.search} src={search}/>
          <input style={styles.input}/>
        </div>
        <div style={styles.sectionsFlex}>
          <img style={styles.home} src={home}/>
          <div>Inicio</div>
        </div>
      </div>
      <div style={styles.settings}>
        <img style={styles.imageSettings} src={settings} />
      </div>
    </div>
  );
};

export default Header;
