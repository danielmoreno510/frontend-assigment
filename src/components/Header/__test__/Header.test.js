import React from "react";
import Adapter from "enzyme-adapter-react-16";
import { shallow, configure } from "enzyme";

import Header from "../Header";

configure({ adapter: new Adapter() });

describe("Header component", () => {
  describe("WHEN the component is render", () => {
    const wrapper = shallow(<Header />);

    it("THEN should display Header component", () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
