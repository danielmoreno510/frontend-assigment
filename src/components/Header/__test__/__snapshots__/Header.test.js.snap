// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component WHEN the component is render THEN should display Header component 1`] = `
<div
  style={
    Object {
      "background": "#1b80c6",
      "color": "white",
      "display": "flex",
      "fontSize": "14px",
      "justifyContent": "space-between",
    }
  }
>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
      }
    }
  >
    <div
      style={
        Object {
          "padding": "12px 48px 12px 32px",
        }
      }
    >
      <img
        src="logo.png"
        style={
          Object {
            "height": "44px",
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "display": "flex",
          "marginRight": "48px",
        }
      }
    >
      <img
        src="search.png"
        style={
          Object {
            "background": "white",
            "borderTopLeftRadius": "10px",
            "height": "24px",
            "marginRight": "2px",
            "padding": "3px",
          }
        }
      />
      <input
        style={
          Object {
            "border": "0px",
            "borderBottomRightRadius": "10px",
            "height": "28px",
            "width": "260px",
          }
        }
      />
    </div>
    <div
      style={
        Object {
          "alignItems": "center",
          "display": "flex",
        }
      }
    >
      <img
        src="home.png"
        style={
          Object {
            "height": "22px",
            "marginRight": "8px",
          }
        }
      />
      <div>
        Inicio
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "alignItems": "center",
        "borderLeft": "1px solid",
        "display": "flex",
        "margin": "10px 0px 10px 0px",
        "padding": "0px 26px 0px 26px",
      }
    }
  >
    <img
      src="settings.png"
      style={
        Object {
          "width": "166px",
        }
      }
    />
  </div>
</div>
`;
